test/fhirtypes/StructureDefinition.test.ts

```diff
@@ -406,6 +406,36 @@ describe('StructureDefinition', () => {
       });
     });
 
+    // Children of primitives
+    it('should set a child of a primitive instance property which has a value', () => {
+      observation.setInstancePropertyByPath('version', 'foo', resolve);
+      observation.setInstancePropertyByPath('version.id', 'bar', resolve);
+      expect(observation.version).toBe('foo');
+      // @ts-ignore
+      expect(observation._version.id).toBe('bar');
+    });
+
+    it('should set a child of a primitive instance property array which has a value', () => {
+      observation.setInstancePropertyByPath('contextInvariant[0]', 'foo', resolve);
+      observation.setInstancePropertyByPath('contextInvariant[0].id', 'bar', resolve);
+      expect(observation.contextInvariant.length).toBe(1);
+      expect(observation.contextInvariant[0]).toBe('foo');
+      // @ts-ignore
+      expect(observation._contextInvariant[0].id).toBe('bar');
+    });
+
+    it('should set a child of a primitive instance property array and null fill the array', () => {
+      observation.setInstancePropertyByPath('contextInvariant[1]', 'foo', resolve);
+      observation.setInstancePropertyByPath('contextInvariant[1].id', 'bar', resolve);
+      expect(observation.contextInvariant.length).toBe(2);
+      expect(observation.contextInvariant[0]).toBeNull();
+      expect(observation.contextInvariant[1]).toBe('foo');
+      // @ts-ignore
+      expect(observation._contextInvariant[0]).toBeNull();
+      // @ts-ignore
+      expect(observation._contextInvariant[1].id).toBe('bar');
+    });
```

Comment:
I didn't modify this test case, but since it also uses `setPropertyByPath`, it should be breakable in a similar way.